sbat.c

```diff
@@ -120,4 +121,168 @@ EFI_STATUS parse_sbat(char *sbat_base, size_t sbat_size, char *buffer,
 	return efi_status;
 }
 
+EFI_STATUS
+verify_single_entry(struct sbat_entry *entry, struct sbat_var *sbat_var_entry)
+{
+        UINT16 sbat_gen, sbat_var_gen;
+
+        if (strcmp(entry->component_name, sbat_var_entry->component_name) == 0) {
+                dprint(L"component %a has a matching SBAT variable entry, verifying\n",
+                       entry->component_name);
+                /* atoi returns zero for failed conversion, so essentially
+                badly parsed component_generation will be treated as zero
+                */
```

Comment:
Right, I remember now reading a comment from @vathpela about this. I wonder if we should add a `strtoul()` to shim or use the one provided by Cryptlib ? The other option is to always start at 1 to make sure that we could discriminate between an error and a 0 component generation number.

@jsetje what do you think ?